api/pkg/backend/service.go

```diff
@@ -447,3 +446,172 @@ func (s *APIService) listStorageType(ctx context.Context, request *restful.Reque
 	storageTypes, err := s.backendClient.ListType(ctx, listTypeRequest)
 	return storageTypes, err
 }
+
+//tiering functions
+func (s *APIService) CreateTier(request *restful.Request, response *restful.Response) {
+	log.Info("Received request for creating tier.")
+	tier := &backend.Tier{}
+	err := request.ReadEntity(&tier)
+	if err != nil {
+		log.Errorf("failed to read request body: %v\n", err)
+		response.WriteError(http.StatusInternalServerError, err)
+		return
+	}
+
+	ctx := common.InitCtxWithAuthInfo(request)
+	actx := request.Attribute(c.KContext).(*c.Context)
+	tier.TenantId = actx.TenantId
+
+	//validation of backends
+	for _, backendId := range tier.Backends {
+		_, err = s.backendClient.GetBackend(ctx, &backend.GetBackendRequest{Id: backendId})
+		if err != nil {
+			log.Errorf("failed to get backendId details of backendId: %v %v\n", backendId, err)
+			response.WriteError(http.StatusInternalServerError, err)
+			return
+		}
+	}
+
+	res, err := s.backendClient.CreateTier(ctx, &backend.CreateTierRequest{Tier: tier})
+	if err != nil {
+		log.Errorf("failed to create tier: %v\n", err)
+		response.WriteError(http.StatusInternalServerError, err)
+		return
+	}
+
+	log.Info("Created backend successfully.")
+	response.WriteEntity(res.Tier)
+
+}
+
+//here backendId can be updated
+func (s *APIService) UpdateTier(request *restful.Request, response *restful.Response) {
+	log.Infof("Received request for updating tier: %v\n", request.PathParameter("id"))
+	id := request.PathParameter("id")
+	updateTierRequest := backend.UpdateTierRequest{Id: id}
+	err := request.ReadEntity(&updateTierRequest)
+	if err != nil {
+		log.Errorf("failed to read request body: %v\n", err)
+		response.WriteError(http.StatusInternalServerError, err)
+		return
+	}
+	ctx := common.InitCtxWithAuthInfo(request)
+
+	//validation of backendId
```

Comment:
multiple backends can not be updated at a time??

Reply:
updated